tasks/initialise.yml

```
@@ -22,27 +22,72 @@
         PGSETUP_INITDB_OPTIONS: >-
           --encoding=UTF8 --locale=en_US.UTF-8 --auth-host=md5
 
-    - name: postgres | postgresql config file
-      template:
+    - name: postgres | check that the postgresql.conf.org file exists
+      ansible.builtin.stat:
+        path: "{{ postgresql_dist_confdir }}/postgresql.conf.org"
+      register: org_file
+
+    - name: postgres | remove the custom configurations from the file, if any
+      command: sed -i '/# Add settings for extensions here/q' "{{ postgresql_dist_confdir }}/postgresql.conf"
+      when:
+        - not org_file.stat.exists
+
+    - name: postgres | remove the Ansible managed line, if any
+      ansible.builtin.lineinfile:
+          path: "{{ postgresql_dist_confdir }}/postgresql.conf"
+          regexp: "#Ansible managed"
+          state: absent
+      when:
+        - not org_file.stat.exists
+
+    - name: postgres | copy a postgresql.conf to postgresql.conf.org
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.copy:
+        src: "{{ postgresql_dist_confdir }}/postgresql.conf"
+        dest: "{{ postgresql_dist_confdir }}/postgresql.conf.org"
+        remote_src: yes
+      check_mode: false
+      when:
+        - not org_file.stat.exists
+
+    # read the default postgresql configuration file
+    - name: postgres | get the postgres conf file contents
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.slurp:
+        src: "{{ postgresql_dist_confdir }}/postgresql.conf.org"
+      register: postgres_config_file_contents_o
+
+    - set_fact: postgres_config_file_contents={{ postgres_config_file_contents_o }}
+
+    - name: postgres | copy postgresql config file
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.template:
         dest: >-
           {{ postgresql_dist_confdir }}/postgresql.conf
         src: "{{ postgresql_dist_conf_postgresql_src }}"
         mode: 0644
+        owner: "{{ postgresql_become_user }}"
       notify:
         - restart postgresql
 
-      become_user: "{{ postgresql_become_user }}"
+    - name: postgres | Remove org config file in check mode and file does not exist
```

```suggestion
    - name: postgres | Remove org config file if not in check mode and file does not exist
```

Really not sure what this sentence was aiming to say, so suggesting ^^^ - but this might not be right - please do check

---

Please revise the formulation ^^^ - I am still not sure my suggestion is right